.github/scripts/trymerge.py

```diff
@@ -310,16 +353,58 @@ def check_if_should_be_merged(pr: GitHubPR, repo: GitRepo) -> None:
             print(f"Skipping rule {rule_name} due to non-matching files: {non_matching_files}")
             continue
         print(f"Matched rule {rule_name} for {pr.pr_num}")
-        return
+        return rule
     raise RuntimeError(f"PR {pr.pr_num} does not match merge rules")
 
 
+def try_revert(repo: GitRepo, pr: GitHubPR, dry_run: bool = False) -> None:
+    def post_comment(msg: str) -> None:
+        gh_post_comment(pr.org, pr.project, pr.pr_num, msg, dry_run=dry_run)
+    if not pr.is_closed():
+        return post_comment(f"Can't revert open PR #{pr.pr_num}")
+    if not RE_REVERT_CMD.match(pr.get_comment_body()):
```

Comment:
What if someone posts a comment in between the revert command and the workflow getting to this point?

Shouldn't there be a search for the last N comments to find the revert command regex, then pass that offset to the below functions?

Reply:
I've considered that, but this adds complexity to the code where it is not, strictly speaking needed. If somebody does that, that author can try again. Or, we can ask probot to pass a comment id as an option. Will do as followup